src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3655,5 +3658,496 @@ private static ImmutableArray<Symbol> GetProperties(CSharpCompilation comp, stri
+            Assert.Equal("System.Int32 X", symbol.ToTestDisplayString());
+            Assert.Equal("C..ctor(System.Int32 X, System.Int32 Y)", symbol.ContainingSymbol.ToTestDisplayString());
+            Assert.Same(symbol.ContainingSymbol, model.GetEnclosingSymbol(x.SpanStart));
+            Assert.Contains(symbol, model.LookupSymbols(x.SpanStart, name: "X"));
+            Assert.Contains("X", model.LookupNames(x.SpanStart));
+
+            x = xs[0];
+            Assert.Equal("Base(X, Y)", x.Parent!.Parent!.Parent!.ToString());
+
+            var symbolInfo = model.GetSymbolInfo(x);
+            Assert.Null(symbolInfo.Symbol);
+            Assert.Empty(symbolInfo.CandidateSymbols);
+            Assert.Equal(CandidateReason.None, symbolInfo.CandidateReason);
+            Assert.Same("<global namespace>", model.GetEnclosingSymbol(x.SpanStart).ToTestDisplayString());
+            Assert.Empty(model.LookupSymbols(x.SpanStart, name: "X"));
+            Assert.DoesNotContain("X", model.LookupNames(x.SpanStart));
+        }
+
+        [Fact]
+        public void BaseArguments_08()
+        {
+            var src = @"
+class Base
+{
+    public Base(int Y)
+    {
+    }
+
+    public Base() {}
+}
+
+record C(int X) : Base(Y)
+{
+    public int Y = 0;
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (11,24): error CS0120: An object reference is required for the non-static field, method, or property 'C.Y'
+                // record C(int X) : Base(Y)
+                Diagnostic(ErrorCode.ERR_ObjectRequired, "Y").WithArguments("C.Y").WithLocation(11, 24)
+                );
+        }
+
+        [Fact]
+        public void BaseArguments_09()
+        {
+            var src = @"
+class Base
+{
+    public Base(int X)
+    {
+    }
+
+    public Base() {}
+}
+
+record C(int X) : Base(this.X)
+{
+    public int Y = 0;
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (11,24): error CS0027: Keyword 'this' is not available in the current context
+                // record C(int X) : Base(this.X)
+                Diagnostic(ErrorCode.ERR_ThisInBadContext, "this").WithLocation(11, 24)
+                );
+        }
+
+        [Fact]
+        public void BaseArguments_10()
+        {
+            var src = @"
+class Base
+{
+    public Base(int X)
+    {
+    }
+
+    public Base() {}
+}
+
+record C(dynamic X) : Base(X)
+{
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (11,27): error CS1975: The constructor call needs to be dynamically dispatched, but cannot be because it is part of a constructor initializer. Consider casting the dynamic arguments.
+                // record C(dynamic X) : Base(X)
+                Diagnostic(ErrorCode.ERR_NoDynamicPhantomOnBaseCtor, "(X)").WithLocation(11, 27)
+                );
+        }
```

Comment:
Do we have a test which demonstrates whether a `partial record` is allowed to reiterate its base class without arguments, e.g.:

```cs
record A(int X) { }

partial record B(int X, int Y) : A(X) { }
partial record B : A { }
```

Reply:
> Do we have a test which demonstrates whether a partial record is allowed to reiterate its base class without arguments, e.g.:

I do not know. I think this is outside of scope of this PR, feel free to add a note to the Test Plan.


---
In reply to: [436069358](https://github.com/dotnet/roslyn/pull/44876#discussion_r436069358) [](ancestors = 436069358)